### PR TITLE
test(data): add unit tests for sequence_padding collator helpers

### PR DESCRIPTION
**Pull request by lonexreb**
# What does this PR do ?

Adds unit tests for `src/megatron/bridge/data/collators/sequence_padding.py`, which currently has no test coverage.

# Changelog

- Add `tests/unit_tests/data/collators/test_sequence_padding.py` (25 tests) covering:
  - `pad_or_truncate_sequence_batch`: pad-to-multiple vs `pad_to_max_length`, truncation, custom `pad_token_id`, labels padded with `IGNORE_INDEX`, `loss_mask` padded with 0, `position_ids` ramp extension, 2D/4D `attention_mask` handling (bool pad value, truncate-then-pad order), `tokens`/`input_ids` mirroring, `sequence_tensor_pad_values` extras, multi-row batches
  - Error branches: missing/1D token tensor, non-positive `sequence_length`, 3D attention mask, and the guard that rejects truncation which would remove every supervised token from a row
  - `use_processor_right_padding`: forces right padding, restores on exit and on exception, tolerates bare tokenizers and processors without `padding_side`

# Verification (contributor-run)

```
$ python -m pytest tests/unit_tests/data/collators/test_sequence_padding.py -q
25 passed
```

Run on a CPU-only environment; the module and tests are pure-CPU tensor logic with no network, CUDA, or checkpoint dependencies. Ruff check and format pass.

# GitHub Actions CI

External contribution — needs `/ok to test <commit-SHA>` from an NVIDIA developer.

# Before your PR is "Ready for review"

- [x] Read and followed Contributor guidelines
- [x] New tests added
- [ ] Documentation — not applicable (tests only)

# Additional Information

- Authored with AI assistance; every assertion was verified against the source module and the full suite was run locally by the submitting contributor.
- Suggested labels: `test` (`area:data`)

### CODE REVIEW

**Comment by yaoyu-33** [comment]
Closing. We are not accepting this contribution. Please do not submit additional AI-generated pull requests to this repository without prior maintainer coordination and thorough, independently verified testing.

**Comment by yaoyu-33** [comment]
Closing. We are not accepting this contribution. Please do not submit additional AI-generated pull requests to this repository without prior maintainer coordination and thorough, independently verified testing.
